pkg/helpers/clustersetmapper.go

```diff
@@ -43,45 +44,67 @@ func (c *ClusterSetMapper) DeleteClusterSet(clusterSetName string) {
 	return
 }
 
-func (c *ClusterSetMapper) GetClustersOfClusterSet(clusterSetName string) sets.String {
-	c.mutex.RLock()
-	defer c.mutex.RUnlock()
+func (c *ClusterSetMapper) DeleteClusterInClusterSet(clusterName string) {
+	if clusterName == "" {
+		return
+	}
 
-	return c.clusterSetToClusters[clusterSetName]
+	c.mutex.Lock()
+	defer c.mutex.Unlock()
+
+	for clusterset, clusters := range c.clusterSetToClusters {
+		if !clusters.Has(clusterName) {
+			continue
+		}
+		clusters.Delete(clusterName)
+		if len(clusters) == 0 {
+			delete(c.clusterSetToClusters, clusterset)
+		}
+	}
+
+	return
 }
+func (c *ClusterSetMapper) UpdateClusterInClusterSet(clusterName, clusterSetName string) {
```

Comment:
one space above, and let's doc how it works.
this func updates clusterset to cluster mapping. if a the clusterset of a cluster is changed, this func remove the previous mapping and  add a new one.